src/legacy/ui/public/courier/fetch/components/shard_failure_modal.tsx

```diff
@@ -0,0 +1,136 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiCodeBlock,
+  EuiTabbedContent,
+  EuiCopy,
+  EuiButton,
+  EuiModalBody,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiCallOut,
+} from '@elastic/eui';
+import { ShardFailureTable } from './shard_failure_table';
+import { ResponseWithShardFailure, Request } from './shard_failure_types';
+
+export interface Props {
+  onClose: () => void;
+  request: Request;
+  response: ResponseWithShardFailure;
+  title: string;
+}
+
+export function ShardFailureModal({ request, response, title, onClose }: Props) {
+  if (!response || !response._shards || !Array.isArray(response._shards.failures) || !request) {
+    // this should never ever happen, but just in case
+    const errorTitle = i18n.translate(
+      'common.ui.courier.fetch.shardsFailedModal.componentErrorTitle',
+      {
+        defaultMessage: 'Sorry, there was an error',
+      }
+    );
+    const errorDesc = i18n.translate(
+      'common.ui.courier.fetch.shardsFailedModal.componentErrorDesc',
+      {
+        defaultMessage: 'The ShardFailureModal component received invalid properties',
+      }
+    );
+    return (
+      <EuiCallOut title={errorTitle} color="danger" iconType="alert">
+        {errorDesc}
+      </EuiCallOut>
+    );
+  }
+
+  const requestJSON = JSON.stringify(request, null, 2);
+  const responseJSON = JSON.stringify(response, null, 2);
+  const failures = response._shards.failures;
+
+  const tabs = [
+    {
+      id: 'table',
+      name: i18n.translate('common.ui.courier.fetch.shardsFailedModal.tabHeaderShardFailures', {
+        defaultMessage: 'Shard failures',
+        description: 'Name of the tab displaying shard failures',
+      }),
+      content: <ShardFailureTable failures={failures} />,
+    },
+    {
+      id: 'json-request',
+      name: i18n.translate('common.ui.courier.fetch.shardsFailedModal.tabHeaderRequest', {
+        defaultMessage: 'Request',
+        description: 'Name of the tab displaying the JSON request',
+      }),
+      content: (
+        <EuiCodeBlock language="json" isCopyable className="shardFailureModal__codeblock">
+          {requestJSON}
+        </EuiCodeBlock>
+      ),
+    },
+    {
+      id: 'json-response',
+      name: i18n.translate('common.ui.courier.fetch.shardsFailedModal.tabHeaderResponse', {
+        defaultMessage: 'Response',
+        description: 'Name of the tab displaying the JSON response',
+      }),
+      content: (
+        <EuiCodeBlock language="json" isCopyable className="shardFailureModal__codeblock">
+          {responseJSON}
+        </EuiCodeBlock>
+      ),
+    },
+  ];
+
+  return (
+    <React.Fragment>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>{String(title)}</EuiModalHeaderTitle>
```

Comment:
`title` is already a string as defined in the props interface. We don't need to convert it to `String` here.

Reply:
done